modules/layers/src/bitmap-layer/bitmap-layer.js

```diff
@@ -213,3 +240,16 @@ export default class BitmapLayer extends Layer {
 
 BitmapLayer.layerName = 'BitmapLayer';
 BitmapLayer.defaultProps = defaultProps;
+
+/**
+ * Decode uv floats from rgb bytes where b contains 4-bit fractions of uv
+ * @param {number[]} color
+ * @returns {number[]} uvs
+ * https://stackoverflow.com/questions/30242013/glsl-compressing-packing-multiple-0-1-colours-var4-into-a-single-var4-variab
+ */
+function unpackUVsFromRGB(color) {
+  const [u, v, fracUV] = color;
+  const vFrac = (fracUV & 0xf0) / 256;
+  const uFrac = (fracUV & 0x0f) / 16;
+  return [(u + uFrac) / 255, (v + vFrac) / 255];
```

Comment:
- This should be `256`
- Use bit shift?

Reply:
Yes I was messing with this to get it right. Ideally this should be implemented separately with tests as part of a `pack` shader module with corresponding JS utils. Without unit tests this is bound to be fragile.